redshift/redshift.go

```diff
@@ -318,12 +282,56 @@ func (r *Redshift) UpdateTable(tx *sql.Tx, targetTable, inputTable Table) error
 		log.Printf("Running command: %s", op)
 		_, err = alterStmt.Exec()
 		if err != nil {
-			return err
+			return fmt.Errorf("issue running statement %s: %s", op, err)
 		}
 	}
 	return nil
 }
 
+// checkSchemas takes in two tables and compares their column schemas to make sure they're compatible.
+// If they have any mismatched columns they are returned in the errors array. If the input table has
+// columns at the end that the target table does not then the appropriate alert tables sql commands are
+// returned.
+func checkSchemas(targetTable, inputTable Table) ([]string, []error) {
+	var columnOps []string
+	var errors []error
+
+	if len(inputTable.Columns) < len(targetTable.Columns) {
+		errors = append(errors, fmt.Errorf("target table has more columns than the input table"))
+	}
+
+	for idx, inCol := range inputTable.Columns {
+		if len(targetTable.Columns) <= idx {
+			fmt.Printf("Missing column -- running alter table\n")
+			alterSQL := fmt.Sprintf(`ALTER TABLE "%s"."%s" ADD COLUMN %s`, targetTable.Meta.Schema, targetTable.Name, getColumnSQL(inCol))
+			columnOps = append(columnOps, alterSQL)
+			continue
```

Comment:
Technically, I think we should have a targetTable.Columns++ here, but it's probably fine.